### ISSUE TITLE
Generic method not showing hover info

### ISSUE THREAD

**Issue by ging-dev** [phpactor/phpactor#1771]
**Problem 1**
```php
<?php

interface B
{
    /**
     * Some document
     */
    public function hello();
}


class A implements B
{
	/**
	 * {@inheritdoc}
	 * 
	 * @return void
	 */
	public function hello()
	{
	}
}

$a = new A();
$a->hello();
```
`hello()` hover:

```
A#hello
Some document
-------------
{} -> incorrect display
```

**Problem 2**

```php
<?php


interface A
{
    /**
     * Some document
     */
    public function hello();
}


/**
 * @template T
 */
class B implements A
{
    /**
     * @return T
     */
    public function hello(): void
    {
    }
}


/** @var B<int> $x */
$x->hello();
```

`hello()` hover doesn't display the document when `B<int>`, `B` without generic work fine.

**Comment by ging-dev** [comment]
> for the second problem, is it rather that `int` is not resolved and it shows`T` instead?
> 
> ![image](https://user-images.githubusercontent.com/530801/179911149-4940ffd6-f019-49d5-a25d-e15c9b3d1f0b.png)

yeah, that's also a problem. But my initial problem was that `hello()` was missing a description from docblock of interface

**Comment by dantleech** [comment]
ok i changed the title of this ticket, I think problem 2 is more important, for problem 1 maybe it can be fixed easily at the same time...

**Comment by dantleech** [comment]
should be fixed in #1826
